lib/moon/helpers/moon_render.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Moon.Helpers.MoonRender do
+  @moduledoc """
+  Functions for rendering Moon components in HEEX/SLIM/... (not Surface) templates
+  """
+
+  import Phoenix.LiveView.Helpers, only: [component: 2, live_component: 1]
+
+  defp get_default_props(module) do
+    Enum.reduce(module.__props__(), %{__context__: %{}}, fn
+      %{name: name, opts: opts}, acc ->
+        Map.put(acc, name, Keyword.get(opts, :default))
+    end)
+  end
+
+  defp transform_slots(
+         props = %{inner_block: [%{__slot__: :inner_block, inner_block: inner_block}]}
+       ) do
+    props
+    |> Map.delete(:inner_block)
+    |> Map.put(:default, [%{__slot__: :default, inner_block: inner_block}])
+  end
+
+  defp transform_slots(props), do: props
+
+  @doc "used for rendering 'dead'/stateless Surface componet"
```

Comment:
Concepts are: (stateless) component and (stateful) live component. Please only agreed terminology.